### PR TITLE
Remove deprecations and deprecate unused CrossEntropySigmoidLoss

### PR DESCRIPTION
**Pull request by adekusar-drl**
### Summary
This PR removes support for "l1" and "l2" loss definitions, use "absolute_error" and "squared_error" instead. Also this PR deprecates `CrossEntropySigmoidLoss` as unused.

PR #362 to be merged first to main and stable.

### CODE REVIEW

File: releasenotes/notes/remove-deprecation-deprecate-sigmoid-319213068ae00ea9.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+deprecations:
+  - |
+    Deprecated class ``CrossEntropySigmoidLoss`` as it is not used in the package.
```
Comment (woodsp-ibm) [inline releasenotes/notes/remove-deprecation-deprecate-sigmoid-319213068ae00ea9.yaml:4]: I am trying to understand this statement. The class exists as a loss function and could be specified on TrainableModel as the loss to be used, so `it is not used in the package` is hard for me to follow, as it seems it could be used and specified as a loss where its applicable for an algorithm.

Comment (adekusar-drl) [reply releasenotes/notes/remove-deprecation-deprecate-sigmoid-319213068ae00ea9.yaml:4]: OK, may be unclear. The loss function is not called anywhere in QML. It could be used, but never used actually. We don't actually need it.

Comment (adekusar-drl) [reply releasenotes/notes/remove-deprecation-deprecate-sigmoid-319213068ae00ea9.yaml:4]: That's right. It can be used externally, but you wrote "is not useful to keep". That's the only reason I want to deprecate it.

**Review by woodsp-ibm — APPROVED** [review]
